llvm/lib/Transforms/Scalar/SimpleLoopUnswitch.cpp

```diff
@@ -2142,34 +2142,33 @@ void visitDomSubTree(DominatorTree &DT, BasicBlock *BB, CallableT Callable) {
 void postUnswitch(Loop &L, LPMUpdater &U, StringRef LoopName,
                   bool CurrentLoopValid, bool PartiallyInvariant,
                   bool InjectedCondition, ArrayRef<Loop *> NewLoops) {
-  // If we did a non-trivial unswitch, we have added new (cloned) loops.
-  if (!NewLoops.empty())
+  auto RecordLoopAsUnswitched = [&](Loop *TargetLoop, StringRef Tag) {
+    auto &Ctx = TargetLoop->getHeader()->getContext();
+    const auto &DisableMDName = (Twine(Tag) + ".disable").str();
```

Comment:
`Twine::str` returns a temporary `std::string`. Can we add a parameter `StringRef DisableTag`  instead?

Reply:
Yeah, the temporary `std::string`, being bound to a `const std::string&` (which extends its lifetime), was a tradeoff in favour of reducing verbosity. Switched to passing `StringRef` as parameter.